homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -0,0 +1,115 @@
+from homeassistant.const import (
+    CONF_PORT, CONF_HOST, CONF_PASSWORD, CONF_USERNAME,
+    STATE_UNKNOWN, CONF_NAME,
+    STATE_ALARM_DISARMED, STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_AWAY)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pythonegardia==1.0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Egardia'
+DEFAULT_PORT = '80'
+DOMAIN = 'egardia'
+NOTIFICATION_ID = 'egardia_notification'
+NOTIFICATION_TITLE = 'Egardia'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
+    })
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Egardia platform."""
+    from pythonegardia import egardiadevice
+    import requests
+    name = config.get(CONF_NAME)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    try:
+        egardiasystem = egardiadevice.EgardiaDevice(host, port,
+                                                    username, password, "")
+    except requests.ConnectionError:
+        _LOGGER.error("Unable to connect to device %s:%s", host, port)
+        return False
+    except egardiadevice.UnauthorizedError:
+        _LOGGER.error("Unable to authorize. Wrong password or username.")
+        return False
+    add_devices([EgardiaAlarm(name, egardiasystem)])
+
+
+class EgardiaAlarm(alarm.AlarmControlPanel):
+    """Representation of a Egardia alarm."""
+
+    def __init__(self, name, egardiasystem):
+        """Initialize object."""
+        self._name = name
+        self._egardiasystem = egardiasystem
+        self._status = STATE_UNKNOWN
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        if self._status == 'ARM':
+            state = STATE_ALARM_ARMED_AWAY
+        elif self._status == 'HOME':
+            state = STATE_ALARM_ARMED_HOME
+        elif self._status == 'DAY HOME':
+            state = STATE_ALARM_ARMED_HOME
+        elif self._status == 'NIGHT HOME':
+            state = STATE_ALARM_ARMED_HOME
+        elif self._status == 'DISARM':
+            state = STATE_ALARM_DISARMED
+        else:
+            state = STATE_UNKNOWN
+        return state
+
+    def getstate(self):
+        """Retrieve the status from the Egardia system."""
+        status = self._egardiasystem.getState()
+        _LOGGER.info("Egardia alarm status: "+status)
+        return status.upper()
+
+    def update(self):
+        """Update the alarm status."""
+        self._status = self.getstate()
```

Comment:
`def getstate(self):` doesn't seem to add much value. I would also like to suggest something like doing a dict lookup.

```python
STATES = {
    'ARM': STATE_ALARM_ARMED_AWAY,
    'HOME': STATE_ALARM_ARMED_HOME,
    'DAY HOME': STATE_ALARM_ARMED_HOME,
  ...
}

    @property
    def state(self):
        """Return the state of the device."""
        return self._state

    def update(self):
        """Update the alarm status."""
        status = self._egardiasystem.getState()
        self._state = ([v for k, v in STATES.items() if status.upper() is k][0])
```